src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/EligibilityAndSuitabilityService.kt

```diff
@@ -0,0 +1,198 @@
+      checksPassed = StatusHelpers.isChecksPassed(eligibilityProgress, suitabilityProgress),
+      eligibility = eligibilityProgress,
+      eligibilityStatus = eligibilityProgress.toStatus(),
+      suitability = suitabilityProgress,
+      suitabilityStatus = suitabilityProgress.toStatus(),
+    )
+  }
+
+  @Transactional
+  fun getEligibilityCriterionView(prisonNumber: String, code: String): EligibilityCriterionView {
+    val offender = offenderRepository.findByPrisonNumber(prisonNumber)
+      ?: throw EntityNotFoundException("Cannot find offender with prisonNumber $prisonNumber")
+
+    val currentAssessment = offender.currentAssessment()
+    val eligibilityProgress = currentAssessment.getEligibilityProgress().dropWhile { it.code != code }.take(2)
+    if (eligibilityProgress.isEmpty()) throw EntityNotFoundException("Cannot find criterion with code $code")
+
+    return EligibilityCriterionView(
+      assessmentSummary = createAssessmentSummary(offender, currentAssessment),
+      criterion = eligibilityProgress[0],
+      nextCriterion = eligibilityProgress.getOrNull(1),
+    )
+  }
+
+  @Transactional
+  fun getSuitabilityCriterionView(prisonNumber: String, code: String): SuitabilityCriterionView {
+    val offender = offenderRepository.findByPrisonNumber(prisonNumber)
+      ?: throw EntityNotFoundException("Cannot find offender with prisonNumber $prisonNumber")
+
+    val currentAssessment = offender.currentAssessment()
+    val suitabilityProgress = currentAssessment.getSuitabilityProgress().dropWhile { it.code != code }.take(2)
+    if (suitabilityProgress.isEmpty()) throw EntityNotFoundException("Cannot find criterion with code $code")
+
+    return SuitabilityCriterionView(
+      assessmentSummary = createAssessmentSummary(offender, currentAssessment),
+      criterion = suitabilityProgress[0],
+      nextCriterion = suitabilityProgress.getOrNull(1),
+    )
+  }
+
+  @Transactional
+  fun saveAnswer(prisonNumber: String, answer: CriterionCheck) {
+    log.info("Saving answer: $prisonNumber, $answer")
+    val offender = offenderRepository.findByPrisonNumber(prisonNumber)
+      ?: throw EntityNotFoundException("Cannot find offender with prisonNumber $prisonNumber")
+    val assessment = offender.currentAssessment()
+
+    val criteriaChecks = assessment.criteriaCheck
+
+    val existingCriteria =
+      criteriaChecks.find { it.criteriaType == CriteriaType.valueOf(answer.type.name) && it.criteriaCode == answer.code }
+
+    val criteria = when {
+      existingCriteria != null -> {
+        criteriaChecks.remove(existingCriteria)
+        existingCriteria.copy(
+          criteriaMet = answer.answers.all { it.value },
+          questionAnswers = answer.answers.toMap(HashMap()),
+          lastUpdatedTimestamp = LocalDateTime.now(),
+        )
+      }
+
+      else -> CriteriaCheck(
+        assessment = assessment,
+        criteriaMet = answer.answers.all { it.value },
```

Comment:
need to inject a strategy to this as one of the suitability is `answers.any { it.value }` rather than `all`